### PR TITLE
Add failing test to reproduce never state issue of #1142

### PR DESCRIPTION
**Pull request by CodingDive**

### CODE REVIEW

File: packages/xstate-react/test/types.test.tsx [hunk 1]
```diff
@@ -114,4 +110,18 @@ describe('useMachine', () => {
 
     render(<YesNo />);
   });
+
+  it('state should not become never after checking state with matches', () => {
+    const YesNo = () => {
+      const [state] = useMachine<any, any>(yesNoMachine);
```
Comment (davidkpiano) [inline packages/xstate-react/test/types.test.tsx:116]: Can you remove the `<any, any>` generics? They're not necessary

Comment (CodingDive) [reply packages/xstate-react/test/types.test.tsx:116]: It was the only way to reproduce the state becoming `never`. Everything seems to work fine in the test when removing the type arguments. I'm also not using `<any, any>` in my code but am running into the same problem.

File: packages/xstate-react/test/types.test.tsx [hunk 2]
```diff
@@ -114,4 +110,18 @@ describe('useMachine', () => {
...
+
+      if (state.matches('no')) {
+        return <span>No</span>;
```
Comment (CodingDive) [inline packages/xstate-react/test/types.test.tsx:119]: The problem seems to happen right when returning right after using `state.matches`. 

Replacing this line with `console.log(state.context.value)` works fine.

In fact, even the following code works fine. 😕 Maybe another TypeScript regression? 

```JavaScript
const isNo = state.matches('no');
if (isNo) {
  return <span>No</span>;
}
```

**Comment by davidkpiano** [comment]
Can you redo this to use `createMachine` instead? `Machine` will be deprecated.

**Comment by CodingDive** [comment]
This particular test is already using `createMachine`. Happened also with `Machine` in my code though.

**Pull request merged**